apps/chat/agent/tools.py

```diff
@@ -81,6 +83,42 @@ def action(
         return "Success"
 
 
+class UpdateScheduledMessageTool(CustomBaseTool):
+    name = "schedule-update-tool"
+    description = "useful to update the schedule of a scheduled message. Use only to update existing schedules"
+    requires_session = True
+    args_schema: type[schemas.ScheduledMessageSchema] = schemas.ScheduledMessageSchema
+
+    def action(
+        self,
+        name: str,
+        weekday: schemas.WeekdaysEnum,
+        hour: int,
+        minute: int,
+        user_specified_custom_date: bool,
+    ):
+        if user_specified_custom_date:
+            return "The user cannot do that. Only weekdays and time of day can be changed"
```

Comment:
Why was this added? Is this to prevent users from trying to change it to some future or past date?

Reply:
Correct. I chose to do this so that users cannot change the date to some date far in the future if they find check-ins annoying

Reply:
👍🏻 Could you add a comment to the code please.

Reply:
https://github.com/dimagi/open-chat-studio/pull/416/commits/9cd8f7a853f7bd23a94614ae5481432d2e8cf2cb